packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -432,34 +432,65 @@ export class FluidDataStoreRuntime
 		return context.getChannel();
 	}
 
-	public createChannel(id: string = uuid(), type: string): IChannel {
+	public addChannel(channel: IChannel): void {
+		const id = channel.id;
 		if (id.includes("/")) {
 			throw new UsageError(`Id cannot contain slashes: ${id}`);
 		}
 
 		this.verifyNotClosed();
 
+		assert(!this.contexts.has(id), "addChannel() with existing ID");
+
+		const type = channel.attributes.type;
+		const factory = this.sharedObjectRegistry.get(channel.attributes.type);
+		if (factory === undefined) {
+			throw new Error(`Channel Factory ${type} not registered`);
+		}
+
+		this.addOrCreateChannelCore(channel);
+	}
+
+	public createChannel(id: string = uuid(), type: string): IChannel {
+		if (id.includes("/")) {
+			throw new UsageError(`Id cannot contain slashes: ${id}`);
+		}
+
+		this.verifyNotClosed();
 		assert(!this.contexts.has(id), 0x179 /* "createChannel() with existing ID" */);
-		this.notBoundedChannelContextSet.add(id);
+
+		const factory = this.sharedObjectRegistry.get(type);
```

Comment:
Still think we should keep this assert from the previous code. The types supposedly don't let this happen, but that was also true before and it was there; maybe there was an edge case or compat case where we could receive undefined.
```suggestion
		assert(type !== undefined, 0x209 /* "Factory Type should be defined" */);
		const factory = this.sharedObjectRegistry.get(type);
```